replicated-security/pion-1/v1.1.0_upgrade.md

```diff
@@ -0,0 +1,245 @@
+cp $(which neutrond) $NEUTRON_HOME/cosmovisor/upgrades/v1.1.0/bin
+```
+
+Then you should get the following structure:
+
+```shell
+.
+├── current -> genesis or upgrades/<name>
+├── genesis
+│   └── bin
+│       └── neutrond  #v1.0.4
+└── upgrades
+    └── v1.1.0
+        └── bin
+            └── neutrond  #v1.1.0
+```
+
+Export the environmental variables:
+
+```shell
+export DAEMON_NAME=neutrond
+# please change to your own neutron home dir
+# please note `DAEMON_HOME` has to be absolute path
+export DAEMON_HOME=$NEUTRON_HOME
+export DAEMON_RESTART_AFTER_UPGRADE=true
+```
+
+Start the node:
+
+```shell
+cosmovisor run  start --x-crisis-skip-assert-invariants --home $DAEMON_HOME
+```
+
+Skipping the invariant checks is strongly encouraged since it decreases the upgrade time significantly and since there are some other improvements coming to the crisis module in the next release of the Cosmos SDK.
+
+#### Expected upgrade result
+
+When the upgrade block height is reached, Neutron will panic and stop:
+
+This may take a few minutes to a few hours.
+After upgrade, the chain will continue to produce blocks when validators with a total sum voting power > 2/3 complete their node upgrades.
+
+### Auto-Downloading the Neutron binary
+
+**This method is not recommended!**
+
+#### Preparation
+
+Install the latest version of Cosmovisor (`1.5.0`):
+
+```shell
+go install cosmossdk.io/tools/cosmovisor/cmd/cosmovisor@latest
+```
+
+Create a cosmovisor folder:
+
+create a cosmovisor folder inside neutron home and move neutrond v1.0.4 into `$NEUTRON_HOME/cosmovisor/genesis/bin`
+
+```shell
+mkdir -p $NEUTRON_HOME/cosmovisor/genesis/bin
+cp $(which neutrond) $NEUTRON_HOME/cosmovisor/genesis/bin
+```
+
+```shell
+.
+├── current -> genesis or upgrades/<name>
+└── genesis
+     └── bin
+        └── neutrond  #v1.0.4
+```
+
+Export the environmental variables:
+
+```shell
+export DAEMON_NAME=neutrond
+# please change to your own neutron home dir
+export DAEMON_HOME=$NEUTRON_HOME
+export DAEMON_RESTART_AFTER_UPGRADE=true
+export DAEMON_ALLOW_DOWNLOAD_BINARIES=true
+```
+
+Start the node:
+
+```shell
+cosmovisor run start --x-crisis-skip-assert-invariants --home $DAEMON_HOME
+```
+
+Skipping the invariant checks can help decrease the upgrade time significantly.
+
+#### Expected result
+
+When the upgrade block height is reached, you can find the following information in the log:
+
+TODO: REPLACE "h" with actual values and make sure "v1.1.0" is the precise version name
+```shell
+ERR UPGRADE "v1.1.0" NEEDED at height: h: upgrade to v1.1.0 and applying upgrade "v1.1.0" at height:h
+```
+
+Then the Cosmovisor will create `$NEUTRON_HOME/cosmovisor/upgrades/v1.1.0/bin` and download the Neuton v1.1.0 binary to this folder according to links in the `--info` field of the upgrade proposal.
+This may take 7 minutes to a few hours, afterwards, the chain will continue to produce blocks once validators with a total sum voting power > 2/3 complete their nodes upgrades.
```

Comment:
according to local tests, it takes ~1hour
stage 1) preparation, no output in logs ~10min
stage 2) applying migration handlers, you can see relevant entities in the logs ~5 min
stage 3) commiting state after migration ~40-50mins